src/cmd/ksh93/tests/pty.sh

```diff
@@ -549,5 +549,17 @@ r ^:test-2: true string\\r\\n$
 !
 done
 
+# err_exit #
+tst $LINENO <<"!"
+L notify job state changes
+
+# 'set -b' should immediately notify the user about job state changes.
+
+p :test-1:
+w set -b; sleep .01 &
+s 20
```

Comment:
As `u` is documented to "read lines from the master until one matches *re*", I think you shouldn't need the `s` (sleep) here; it should keep reading/waiting until reaching either a match or EOF.